src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -233,6 +243,58 @@ public void jobFinished() {
             tablesToMerge.addAll(Arrays.asList(GtfsPlusTable.tables));
         }
         int numberOfTables = tablesToMerge.size();
+        // Before initiating the merge process, run some pre-processing to check for id conflicts for certain tables
+        if (mergeType.equals(SERVICE_PERIOD)) {
+            mergeStrategy = getMergeStrategy(feedsToMerge);
+            if (mergeStrategy == CHECK_STOP_TIMES) {
+                Feed futureFeed = new Feed(DataManager.GTFS_DATA_SOURCE, feedsToMerge.get(0).version.namespace);
+                Feed pastFeed = new Feed(DataManager.GTFS_DATA_SOURCE, feedsToMerge.get(1).version.namespace);
+                for (String tripId : intersectingTripIds) {
+                    // Fetch all ordered stop_times for each common trip_id, hash them, and compare the two sets for the
+                    // future and current feed. If the stop_times are an exact match, include one instance of the trip
+                    // (ignoring the other identical one). If they do not match, modify the current trip_id and include.
+                    List<StopTime> futureStopTimes = Lists.newArrayList(futureFeed.stopTimes.getOrdered(tripId));
+                    List<StopTime> pastStopTimes = Lists.newArrayList(pastFeed.stopTimes.getOrdered(tripId));
```

Comment:
It seems rather memory-inefficient to put everything into a list upfront. See more comments in the stopTimesMatch method.

Reply:
Ah, actually, looking at the internals this is probably fine since it's only getting the stop times for a particular trip.